common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from django.utils import timezone
+
+from django_filters.rest_framework import DjangoFilterBackend
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from student.models import CourseEnrollment
+from entitlements.filters import CourseEntitlementFilter
+from entitlements.models import CourseEntitlement
+from entitlements.serializers import CourseEntitlementSerializer
+
+log = logging.getLogger("edx.entitlement")
+
+
+class EntitlementViewSet(viewsets.ModelViewSet):
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, permissions.IsAdminUser,)
+    queryset = CourseEntitlement.objects.all()
```

Comment:
Performance question.  Since we know we will be serializing the username as well should this `select_related` user?